homeworks/08-cats-catcher/src/App/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { FC, useState, useEffect, useRef } from "react";
+import { of, interval, fromEvent, combineLatest } from "rxjs";
+import { map, filter } from "rxjs/operators";
+import random from "lodash/random";
+
+import { ReactComponent as WindowIcon } from "assets/icons/window.svg";
+import { ReactComponent as CatIcon } from "assets/icons/kitty.svg";
+import { Container, IconWrapper, ControllersContainer } from "./App.styled";
+import { useGenerateWindowsPositions } from "./generateFiled";
+
+const App: FC = () => {
+  const catRef = useRef<HTMLSpanElement>(null);
+  const resetButtonRef = useRef<HTMLButtonElement>(null);
+  const { positions, onReset } = useGenerateWindowsPositions();
+  const [catPosition, setCatPosition] = useState<null | number[]>(null);
+  const [score, setScore] = useState(0);
+
+  useEffect(() => {
+    const positions$ = of(positions);
+    const interval$ = interval(1000);
+
+    const cat$ = combineLatest([positions$, interval$])
+      .pipe(
+        filter(([pos]) => {
+          return !!pos;
+        }),
+        map(([pos]) => pos![random(0, 9)])
+      )
+      .subscribe((val) => setCatPosition(val));
+
+    const shoot$ = fromEvent(catRef.current!, "click").subscribe(() =>
+      setScore((prevState) => prevState + 1)
+    );
+
+    return () => {
+      cat$.unsubscribe();
+      shoot$.unsubscribe();
+    };
+  }, [positions]);
```

Comment:
I think hooks cannot allow you to feel all powerful of RxJs because of subscribing/unsubscribing every time when something is changed. Reactive code can react on changes by itself.
However in general everything looks good.